tests/infra_test.go

```diff
@@ -298,14 +297,18 @@ var _ = Describe("Infrastructure", func() {
 		}, 300)
 
 		It("should include VMI infos for a running VM", func() {
-			vmi, nodeName, _, metrics := newRandomVMIWithMetrics(virtClient, "kubevirt_vm_")
+			vmis, metrics := newRandomVMIWithMetrics(virtClient, "kubevirt_vm_")
+			Expect(vmis).To(HaveLen(1))
+
 			By("Checking the collected metrics")
 			keys := getKeysFromMetrics(metrics)
 			for _, key := range keys {
-				// we don't care about the ordering of the labels
-				Expect(key).To(ContainSubstring(fmt.Sprintf("node=%s", nodeName)))
-				Expect(key).To(ContainSubstring(fmt.Sprintf("namespace=%s", vmi.Namespace)))
-				Expect(key).To(ContainSubstring(fmt.Sprintf("name=%s", vmi.Name)))
+				for _, vmi := range vmis {
+					// we don't care about the ordering of the labels
+					Expect(key).To(ContainSubstring(fmt.Sprintf("node=%s", vmi.Status.NodeName)))
```

Comment:
Use composition Expect(key).To(SatisfyAll(Contain..., Contain...)

Reply:
This is an old patch, already fixed with committed code.